KIC 2.0: handle stderrthreshold flag

Follow up from https://github.com/Kong/kubernetes-ingress-controller/pull/1294#pullrequestreview-655935956

Although the controller's logger doesn't make use of the `--stderrthreshold` flag, its client-go instance uses klog and does. However, Cobra doesn't play nicely with it; we need to figure out how to make it handle the klog flags properly.

Hmm, so maybe not as easy as I thought. https://github.com/Kong/kubernetes-ingress-controller/commit/852a8f50684ceb96df066580a4768513712f09c8 is based off a combination of
https://flowerinthenight.com/blog/2019/02/05/golang-cobra-klog
https://pkg.go.dev/github.com/spf13/pflag#readme-supporting-go-flags-when-using-pflag
https://pkg.go.dev/k8s.io/klog#readme-how-to-use-klog

But this results in a conflict:

```
$ ./bin/manager --help
 flag redefined: kubeconfig
panic:  flag redefined: kubeconfig
```
Removing our kubeconfig definition still overwrites Cobra's generated help with the klog flag set's help, and I'm not sure how we pull out the kubeconfig flag for use with controller-runtime init :(

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -42,6 +42,12 @@
   [#1451](https://github.com/Kong/kubernetes-ingress-controller/pull/1451)
   [#1509](https://github.com/Kong/kubernetes-ingress-controller/pull/1509)
 
+#### Under the hood
+
+- The historical `--stderrthreshold` flag is now deprecated: it no longer has
+  any effect when used and will be removed in a later release.
+  [#1297](https://github.com/Kong/kubernetes-ingress-controller/issues/1297)
+
 ## [2.0.0-alpha.2] - 2021/07/07
 
 #### Fixed

--- a/railgun/pkg/config/config.go
+++ b/railgun/pkg/config/config.go
@@ -119,11 +119,6 @@ func (c *Config) FlagSet() *pflag.FlagSet {
 	flagSet.StringVar(&c.MetricsAddr, "metrics-bind-address", fmt.Sprintf(":%v", MetricsPort), "The address the metric endpoint binds to.")
 	flagSet.StringVar(&c.ProbeAddr, "health-probe-bind-address", fmt.Sprintf(":%v", HealthzPort), "The address the probe endpoint binds to.")
 	flagSet.StringVar(&c.KongAdminURL, "kong-admin-url", "http://localhost:8001", `The Kong Admin URL to connect to in the format "protocol://address:port".`)
-	flagSet.Float32Var(&c.ProxySyncSeconds, "sync-rate-limit", proxy.DefaultSyncSeconds,
-		fmt.Sprintf(
-			"Define the rate (in seconds) in which configuration updates will be applied to the Kong Admin API. (default: %g seconds) (DEPRECATED, use --proxy-sync-seconds instead)",
-			proxy.DefaultSyncSeconds,
-		))
 	flagSet.Float32Var(&c.ProxySyncSeconds, "proxy-sync-seconds", proxy.DefaultSyncSeconds,
 		fmt.Sprintf(
 			"Define the rate (in seconds) in which configuration updates will be applied to the Kong Admin API. (default: %g seconds)",
@@ -184,6 +179,14 @@ func (c *Config) FlagSet() *pflag.FlagSet {
 
 	// Misc
 	flagSet.BoolVar(&c.EnableProfiling, "profiling", false, "Enable profiling via web interface host:10256/debug/pprof/")
+
+	// Deprecated (to be removed in future releases)
+	flagSet.Float32Var(&c.ProxySyncSeconds, "sync-rate-limit", proxy.DefaultSyncSeconds,
+		fmt.Sprintf(
+			"Define the rate (in seconds) in which configuration updates will be applied to the Kong Admin API. (default: %g seconds) (DEPRECATED, use --proxy-sync-seconds instead)",
+			proxy.DefaultSyncSeconds,
+		))
+	flagSet.Int("stderrthreshold", 0, "DEPRECATED: has no effect and will be removed in future releases (see github issue #1297)")
 
 	return &flagSet.FlagSet
 }

